src/update/update.h

```diff
@@ -41,295 +39,323 @@ static const int32_t VERSIONBITS_TOP_MASK = 0xE0000000UL;
+    /// @param update Update being evaluated.
+    /// @return Cumulative voting results for the last votingPeriod rounds.
+    virtual VoteResult GetVote(const CBlockIndex *blockIndex, const Update &update) = 0;
 };
 
-class MinerRoundVoting : public IRoundVoting
-{
-   public :
-      MinerRoundVoting() {};
-      virtual ~MinerRoundVoting() {};
+class MinerRoundVoting : public IRoundVoting {
+public :
+    MinerRoundVoting() {};
 
-      VoteResult GetVote(const CBlockIndex* blockIndex, const Update& update) override;
+    virtual ~MinerRoundVoting() {};
 
-   private :
-      std::map<UpdateCacheKey, VoteResult> cache;
+    VoteResult GetVote(const CBlockIndex *blockIndex, const Update &update) override;
+
+private :
+    std::map <UpdateCacheKey, VoteResult> cache;
 };
 
-class NodeRoundVoting : public IRoundVoting
-{
-   public :
-      NodeRoundVoting() {};
-      virtual ~NodeRoundVoting() {};
+class NodeRoundVoting : public IRoundVoting {
+public :
+    NodeRoundVoting() {};
+
+    virtual ~NodeRoundVoting() {};
 
-      VoteResult GetVote(const CBlockIndex* blockIndex, const Update& update) override;
+    VoteResult GetVote(const CBlockIndex *blockIndex, const Update &update) override;
 
-   private :
-      std::map<UpdateCacheKey, VoteResult> cache;
+private :
+    std::map <UpdateCacheKey, VoteResult> cache;
 };
 
-class MinerUpdateVoting : public IUpdateVoting
-{
-   public :
-      MinerUpdateVoting(MinerRoundVoting* minerRoundVoting) : minerRoundVoting(minerRoundVoting) {};
-      virtual ~MinerUpdateVoting() {};
+class MinerUpdateVoting : public IUpdateVoting {
+public :
+    MinerUpdateVoting(MinerRoundVoting *minerRoundVoting) : minerRoundVoting(minerRoundVoting) {};
 
-      VoteResult GetVote(const CBlockIndex* blockIndex, const Update &update) override;
+    virtual ~MinerUpdateVoting() {};
 
-    protected:
-      MinerRoundVoting* minerRoundVoting;
+    VoteResult GetVote(const CBlockIndex *blockIndex, const Update &update) override;
+
+protected:
+    MinerRoundVoting *minerRoundVoting;
 };
 
-class NodeUpdateVoting : public IUpdateVoting
-{
-   public :
-      NodeUpdateVoting(NodeRoundVoting* nodeRoundVoting) : nodeRoundVoting(nodeRoundVoting) {};
-      virtual ~NodeUpdateVoting() {};
+class NodeUpdateVoting : public IUpdateVoting {
+public :
+    NodeUpdateVoting(NodeRoundVoting *nodeRoundVoting) : nodeRoundVoting(nodeRoundVoting) {};
+
+    virtual ~NodeUpdateVoting() {};
 
-      VoteResult GetVote(const CBlockIndex* blockIndex, const Update &update) override;
+    VoteResult GetVote(const CBlockIndex *blockIndex, const Update &update) override;
 
-    protected:
-      NodeRoundVoting* nodeRoundVoting;
+protected:
+    NodeRoundVoting *nodeRoundVoting;
 };
 
-typedef struct StateInfo
-{
-   EUpdateState State;
-   int64_t      FinalHeight;    // Active height when state in (LockedIn, Active), Failed height when state == Failed
+typedef struct VoteStats {
+    VoteResult *nodeVoteResult;
```

Comment:
How is this memory managed?
nodeVoteResult is not specific enough - should be nodeUpdateVoteResult or nodeRoundVoteResult?